client/utils/currency/index.js

```diff
@@ -64,6 +64,70 @@ export const formatCurrency = ( amount, currencyCode = 'USD' ) => {
 	}
 };
 
+/**
+ * Formats exchange rate string from one currency to another.
+ *
+ * @param {Object} from          Source currency and amount for exchange rate calculation.
+ * @param {string} from.currency Source currency code.
+ * @param {number} from.amount   Source amount.
+ * @param {Object} to            Target currency and amount for exchange rate calculation.
+ * @param {string} to.currency   Target currency code.
+ * @param {number} to.amount     Target amount.
+ *
+ * @return {string?} formatted string like `€1,00 → $1,19: $29.99`.
+ *
+ * */
+export const formatFX = ( from, to ) => {
+	if ( ! from.currency || ! to.currency ) {
+		return;
+	}
+
+	const fromAmount = isZeroDecimalCurrency( from.currency ) ? 1 : 100;
+	return `${ formatCurrency(
+		fromAmount,
+		from.currency
+	) } → ${ formatExchangeRate( from, to ) }: ${ formatCurrency(
+		Math.abs( to.amount ),
+		to.currency
+	) }`;
+};
+
+function formatExchangeRate( from, to ) {
+	let exchangeRate =
+		'number' === typeof to.amount &&
+		'number' === typeof from.amount &&
+		0 !== from.amount
+			? Math.abs( to.amount / from.amount )
+			: 0;
+	if ( isZeroDecimalCurrency( to.currency ) ) {
+		exchangeRate *= 100;
+	}
+
+	if ( isZeroDecimalCurrency( from.currency ) ) {
+		exchangeRate /= 100;
+	}
+
+	const exchangeCurrencyConfig = find( currencyData, {
+		code: to.currency.toUpperCase(),
+	} );
+
+	const precision = 1 > exchangeRate ? 6 : 5;
+	const isZeroDecimal = isZeroDecimalCurrency( to.currency );
+
+	if ( ! exchangeCurrencyConfig ) {
+		sprintf(
```

Comment:
Was this meant to be assigned / returned?

Reply:
Great catch, @dechov ! I think it should be early return with fallback formatting. Weird it's not caught by tests.
And sorry for introducing this.

Reply:
It's totally inert code, and I didn't even look into it enough to know what bug it might _not_ be averting :-) so no need to apologize, I'd just spotted what looked like some extraneous lines.